### ISSUE TITLE
Upstream delaying of diagnostics about TLS declarations for device 

### ISSUE THREAD

**Issue by Fznamznon** [intel/llvm#946]
Thread-local storage is not supported for device, so we need to make corresponding diagnostic deferred.
Without it, compiler is emitting diagnostics from standard C++ headers during device code compilation:
```
/usr/lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/mutex:650:10: error: thread-local storage is not supported for the current target
  extern __thread void* __once_callable;
```
We made it with the following commit:
```
commit ddc1a7f3c364ebb951bbe2b379e26f87244300bd
Author: Premanand M Rao <premanand.m.rao@intel.com>
Date:   Mon Aug 26 07:57:46 2019 -0700

    [SYCL] Do not emit errors for TLS declarations

    Use of TLS variables in SYCL code is diagnosed instead.

    Signed-off-by: Premanand M Rao <premanand.m.rao@intel.com>
```

**Comment by Fznamznon** [comment]
Diagnostic moved to regular Sema https://github.com/intel/llvm/commit/eb373c497efbdbaefff58537f3e4774f8ca2aff1

**Comment by Fznamznon** [comment]
Since https://reviews.llvm.org/D74387 added infrastructure for SYCL deferred diagnostics, we can upstream https://github.com/intel/llvm/commit/eb373c497efbdbaefff58537f3e4774f8ca2aff1 as well, so it will help to compile basic SYCL code samples.
@erichkeane, since you are an author of https://github.com/intel/llvm/commit/eb373c497efbdbaefff58537f3e4774f8ca2aff1, I'd like to ask you to upstream it. Do you have some bandwitdth?
NOTE: Before putting it to review, I'd double check if OpenMP (or other offloading programming models) does same things somewhere and if it does, then approach should be generalized.

**Comment by erichkeane** [comment]
I have absolutely zero bandwidth in the near future, but I can put it on my list of 'things to do'... Not sure when I'll be able to get to it.

**Comment by Fznamznon** [comment]
@erichkeane , I don't get what you mean. Why this is the 'right' way to fix?
AFAIK we use standard targets with `sycldevice` environment. Do you mean we need to upstream `sycldevice` triple environment?

**Comment by Fznamznon** [comment]
https://reviews.llvm.org/D81641

**Comment by Fznamznon** [comment]
Merged.
